warehouse/query-core/src/main/java/datawave/query/iterator/logic/TermFrequencyIndexIterator.java

```diff
@@ -111,8 +121,6 @@ public TermFrequencyIndexIterator(Key fiStartKey, boolean startInclusive, Key fi
         
         this.includeColumnFamilies = true;
         
-        this.document = new Document();
```

Comment:
I suspect that this instance of `Document` was created here so that the public `document()` method never returns `null`. After this change, the `document()` method will return `null` if called prior to the `next()` method. 

As a result, we might want to leave it in place unless you have a really clear idea about everywhere (and when) that method is being called. Although not ideal, the overhead of a single document object is likely negligible.

Reply:
What do you think about leaving this code as is and modifying the document method to look like this?
```
public Document document(){
  if(document == null){
    document = new Document()
  }
  return document;
}
```